src/analysis/normed_space/multilinear.lean

```diff
@@ -0,0 +1,869 @@
+  simpa
+end
+
+variables (𝕜 E E₂)
+
+/-- The space of continuous multilinear maps on `Π(i : fin (n+1)), E i` is canonically isomorphic to
+the space of continuous multilinear maps on `Π(i : fin n), E i.succ` with values in the space of
+continuous linear maps on `E 0`, by separating the first variable. We register this isomorphism as a
+linear isomorphism in `continuous_multilinear_curry_right_equiv_aux 𝕜 E E₂`.
+The algebraic version (without continuity assumption on the maps) is
+`multilinear_curry_right_equiv 𝕜 E E₂`, and the topological isomorphism (registering
+additionally that the isomorphism is continuous) is
+`continuous_multilinear_curry_right_equiv 𝕜 E E₂`.
+
+The direct and inverse maps are given by `f.uncurry_right` and `f.curry_right`. Use these
+unless you need the full framework of linear equivs. -/
+def continuous_multilinear_curry_right_equiv_aux :
+  (continuous_multilinear_map 𝕜 (λ(i : fin n), E i.succ) (E 0 →L[𝕜] E₂)) ≃ₗ[𝕜]
+  (continuous_multilinear_map 𝕜 E E₂) :=
+{ to_fun    := continuous_multilinear_map.uncurry_right,
+  add       := λf₁ f₂, by { ext m, refl },
+  smul      := λc f, by { ext m, refl },
+  inv_fun   := continuous_multilinear_map.curry_right,
+  left_inv  := continuous_multilinear_map.curry_uncurry_right,
+  right_inv := continuous_multilinear_map.uncurry_curry_right }
+
+/-- The space of continuous multilinear maps on `Π(i : fin (n+1)), E i` is canonically isomorphic to
+the space of continuous multilinear maps on `Π(i : fin n), E i.succ` with values in the space of
+continuous linear maps on `E 0`, by separating the first variable. We register this isomorphism in
+`continuous_multilinear_curry_right_equiv 𝕜 E E₂`. The algebraic version (without topology) is given
+in `multilinear_curry_right_equiv 𝕜 E E₂`.
+
+The direct and inverse maps are given by `f.uncurry_right` and `f.curry_right`. Use these
+unless you need the full framework of continuous linear equivs. -/
+def continuous_multilinear_curry_right_equiv :
+  (continuous_multilinear_map 𝕜 (λ(i : fin n), E i.succ) (E 0 →L[𝕜] E₂)) ≃L[𝕜]
+  (continuous_multilinear_map 𝕜 E E₂) :=
+{ continuous_to_fun := begin
+    refine (continuous_multilinear_curry_right_equiv_aux 𝕜 E E₂).to_linear_map.continuous_of_bound
+      (1 : ℝ) (λf, le_of_eq _),
+    rw one_mul,
+    exact f.uncurry_right_norm
+  end,
+  continuous_inv_fun := begin
+    refine (continuous_multilinear_curry_right_equiv_aux 𝕜 E E₂).symm.to_linear_map.continuous_of_bound
+      (1 : ℝ) (λf, le_of_eq _),
+    rw one_mul,
+    exact f.curry_right_norm
+  end,
+  .. continuous_multilinear_curry_right_equiv_aux 𝕜 E E₂ }
+
+/-!
+### Currying with `0` variables
+
+The space of multilinear maps with `0` variables is trivial. Therefore, the space of continuous
+multilinear maps on `(fin 0) → G` with values in `E₂` is isomorphic (and even isometric) to `E₂`.
+As this is the zeroth step in the construction of iterated derivatives, we register this
+isomorphism. -/
```

Comment:
Probably it's better to say explicitly what "trivial" means here. I spent some time realizing that `multilinear_map.map_zero` needs `[nonempty ι]`, and in this case we can send zero to any element.

Reply:
Done